subprojects/core/src/test/groovy/org/gradle/api/tasks/JavaExecTest.groovy

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright 2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.gradle.api.tasks
+
+import org.gradle.api.internal.AbstractTask
+import org.gradle.process.ExecResult
+import org.gradle.process.internal.ExecAction
+import org.gradle.process.internal.ExecException
+
+class JavaExecTest extends AbstractTaskTest {
+    JavaExec execTask
+    def execAction = Mock(ExecAction) {
+        getArgumentProviders() >> []
+    }
+
+    def setup() {
+        execTask = createTask(JavaExec.class)
+        execTask.setExecAction(execAction)
+    }
+
+    AbstractTask getTask() {
+        return execTask
+    }
+
+    def "executes action on execute"() {
+        when:
+        execTask.setExecutable("ls")
```

Comment:
⭕️ Nitpick: We should use an executable that is more Java-y, maybe just `java` would better show the intent here.